app/src/main/java/it/quip/android/activity/ShareQuipActivity.java

```diff
@@ -1,39 +1,66 @@
 package it.quip.android.activity;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.view.Menu;
-import android.view.MenuItem;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+import java.util.List;
 
 import it.quip.android.R;
+import it.quip.android.fragment.CirclesSelectListFragment;
+import it.quip.android.model.Circle;
 
 public class ShareQuipActivity extends AppCompatActivity {
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    public static final int SHARE_QUIP_RESULT = 1111;
+    public static final String SELECTED_CIRCLES = "SELECTED_CIRCLES";
+
+    private Button btPost;
+    private CirclesSelectListFragment mCirclesSelectListFragment;
+
+    private static List<Circle> sCircles;
+
+    private void setupDependencies() {
+        mCirclesSelectListFragment = CirclesSelectListFragment.newInstance();
+    }
+
+    private void setupView() {
         setContentView(R.layout.activity_share_quip);
+        btPost = (Button) findViewById(R.id.bt_share_quip_post);
+        btPost.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                sCircles = mCirclesSelectListFragment.getSelectedValues();
+                finish();
+                overridePendingTransition(R.anim.slide_out_right, R.anim.slide_in_left);
+            }
+
+        });
     }
 
-    @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
-        // Inflate the menu; this adds items to the action bar if it is present.
-        getMenuInflater().inflate(R.menu.menu_share_quip, menu);
-        return true;
+    private void showCircleSelectFragment() {
+        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
+        ft.replace(R.id.fl_share_quip_circle_picker, mCirclesSelectListFragment);
+        ft.commit();
     }
 
     @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
-        // Handle action bar item clicks here. The action bar will
-        // automatically handle clicks on the Home/Up button, so long
-        // as you specify a parent activity in AndroidManifest.xml.
-        int id = item.getItemId();
-
-        //noinspection SimplifiableIfStatement
-        if (id == R.id.action_settings) {
-            return true;
-        }
-
-        return super.onOptionsItemSelected(item);
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setupDependencies();
+        setupView();
+        showCircleSelectFragment();
+    }
+
+    public static List<Circle> getCircles() {
+        List<Circle> result = sCircles;
+        sCircles = null;
```

Comment:
Just to ensure there is not errant dangling pointer to old data. Its a static reference so I want to be sure it doesn't pick up old data the next time its used.